component.go

```diff
@@ -0,0 +1,52 @@
+package jira
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+)
+
+// ComponentService handles components for the JIRA instance / API.
+//
+// JIRA API docs: https://docs.atlassian.com/software/jira/docs/api/REST/7.10.1/#api/2/component
+type ComponentService struct {
+	client *Client
+}
+
+// CreateComponentOptions are passed to the ComponentService.Create function to create a new JIRA component
+type CreateComponentOptions struct {
+	Name         string `json:"name,omitempty" structs:"name,omitempty"`
+	Description  string `json:"description,omitempty" structs:"description,omitempty"`
+	Lead         *User  `json:"lead,omitempty" structs:"lead,omitempty"`
+	LeadUserName string `json:"leadUserName,omitempty" structs:"leadUserName,omitempty"`
+	AssigneeType string `json:"assigneeType,omitempty" structs:"assigneeType,omitempty"`
+	Assignee     *User  `json:"assignee,omitempty" structs:"assignee,omitempty"`
+	Project      string `json:"project,omitempty" structs:"project,omitempty"`
+	ProjectID    int    `json:"projectId,omitempty" structs:"projectId,omitempty"`
+}
+
+// Create creates a new JIRA component based on the given options.
+func (s *ComponentService) Create(options *CreateComponentOptions) (*ProjectComponent, *Response, error) {
+	apiEndpoint := "rest/api/2/component"
+	req, err := s.client.NewRequest("POST", apiEndpoint, options)
+	if err != nil {
+		return nil, nil, err
+	}
+	resp, err := s.client.Do(req, nil)
+	if err != nil {
+		// incase of error return the resp for further inspection
+		return nil, resp, err
+	}
+
+	component := new(ProjectComponent)
```

Comment:
Is there a special reason that you're doing all of the deserialization yourself? 

`Do` accepts an empty struct so you can pass `component` in.

```go
component := new(ProjectComponent)
resp, err := s.client.Do(req, component)
if err != nil {
  err = NewJiraError(resp, err)
}
return component, resp, err
```

Reply:
I was just trying to follow conventions I saw in other places. I figured there was some reason that the serialization was being done this way, for example in `Issue.Create` https://github.com/natemara/go-jira/blob/bd60d4e7dad1573ff3e0886aee5ad82af70c865d/issue.go#L626. If this is not the recommended way to do deserialization, I can fix it.